src/cache/precipitation_precomputed_quantities.jl

```diff
@@ -1,20 +1,57 @@
 #####
-##### Precomputed quantities
+##### Precomputed quantities for precipitation processes
 #####
-import CloudMicrophysics.Microphysics1M as CM1
+
 import CloudMicrophysics.MicrophysicsNonEq as CMNe
+import CloudMicrophysics.Microphysics1M as CM1
+
+import Thermodynamics as TD
+import ClimaCore.Operators as Operators
+import ClimaCore.Fields as Fields
+
+const Iₗ = TD.internal_energy_liquid
+const Iᵢ = TD.internal_energy_ice
 
 """
-    set_precipitation_precomputed_quantities!(Y, p, t)
+   Kin(ᶜw_precip, ᶜu_air)
+
+    - ᶜw_precip - teminal velocity of cloud consensate or precipitation
+    - ᶜu_air - air velocity
+
+Helper function to compute the kinetic energy of cloud condensate and
+precipitation.
+"""
+function Kin(ᶜw_precip, ᶜu_air)
+    return @. lazy(
+        norm_sqr(
+            Geometry.UVWVector(0, 0, -(ᶜw_precip)) + Geometry.UVWVector(ᶜu_air),
+        ) / 2,
+    )
+end
 
-Updates the precipitation terminal velocity stored in `p`
-for the 1-moment microphysics scheme
 """
-function set_precipitation_precomputed_quantities!(Y, p, t)
-    @assert (p.atmos.precip_model isa Microphysics1Moment)
+    set_precipitation_velocities!(Y, p, moisture_model, precip_model)
 
-    (; ᶜwᵣ, ᶜwₛ) = p.precomputed
+Updates the precipitation terminal velocity, cloud sedimentation velocity,
+and their contribution to total water and energy advection.
+"""
+function set_precipitation_velocities!(Y, p, _, _)
+    (; ᶜwₜqₜ, ᶜwₕhₜ) = p.precomputed
+    @. ᶜwₜqₜ = Geometry.WVector(0)
+    @. ᶜwₕhₜ = Geometry.WVector(0)
```

Comment:
In this case, where are these values used?

Reply:
Yeah. They are used in https://github.com/CliMA/ClimaAtmos.jl/blob/aca95c135fd0cbd3722e5becd4a2f15a94e061d0/src/prognostic_equations/remaining_tendency.jl#L21

But good point. I'll write two versions of that function and then not need the zeros

Reply:
You may need to write two versions for the implicit solver too, if we ever make it implicit. But that should be doable, just a bit more work.

Reply:
(and if it's too much work, I'm fine with just setting it zero)

Reply:
Then maybe lets keep it as is for now. And once we settle on whether this should be implicit or explicit I can simplify it.